### PR TITLE
[Stats Refresh] Insights detail view async loading

### LINKED ISSUE #12424: [Stats Refresh] Insights: Load overview block independently in details view
On Insights we should load all the details view using the ghost animation

### PR DESCRIPTION
**Pull request by danielebogo**
Fixes #12424 

This PR removes the NRV from the Insights detail view and uses the async loading + ghost view approach instead.
The section affected are:
• _Followers_
• _Comments_
• _Tags & Categories_
• _This Year_

## To test:
• Select Stats from a site with data (en.blog is a good candidate) and select _Insights_
• Make sure the sections mentioned above are present in _Insights_
• For every section click on _View more_ and open the detail
• You should see the Ghost views before the content is loaded

Update release notes:

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
FWIW, https://github.com/wordpress-mobile/WordPress-iOS/pull/11881 disabled showing 'View more' for _Comments_, so that is untestable.

**Comment by ScoutHarris** [comment]
Hey @danielebogo .

When loading _This Year_ details, there is a grey gap at the top where the date bar is supposed to be. It looks kinda strange.

Loading:
this_year_loading

After loading:
this_year_loaded

File: WordPress/Classes/Stores/StatsStore+Cache.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+extension StatsInsightsStore: StatsStoreCacheable {
+    func containsCachedData(for type: InsightType) -> Bool {
+        switch type {
+        case .latestPostSummary:
+            return state.lastPostInsight != nil
+        case .allTimeStats:
+            return state.allTimeStats != nil
+        case .followersTotals:
```
Comment (ScoutHarris) [inline WordPress/Classes/Stores/StatsStore+Cache.swift]: Is this necessary, since `default` returns `false`?

**Comment by danielebogo** [comment]
@ScoutHarris thanks for the review! Header and enumeration are ok now! Ready for another look!

**Pull request merged**